src/main/java/org/openmrs/addonindex/scheduled/FetchDetailsToIndex.java

```diff
@@ -109,6 +109,7 @@ void getDetailsAndIndex(AddOnToIndex toIndex) throws Exception {
 			}
 			
 			if (fetchExtraDetails) {
+				handler.fetchDownloadCounts(toIndex, infoAndVersions);
```

Comment:
assuming you've added a new subinterface like I suggest above, this would become:

    if (handler instanceof SupportsDownloadCounts) {
        hander.fetchDownloadCounts(...);
    }